railties/lib/rails/generators/app_base.rb

```diff
@@ -422,6 +427,29 @@ def os_supports_listen_out_of_the_box?
         /darwin|linux/.match?(RbConfig::CONFIG["host_os"])
       end
 
+      def target_rails_prerelease(self_command = "new")
+        gemfile = "rails_prerelease.gemfile"
+
+        if rails_prerelease? && !ENV["BUNDLE_GEMFILE"]&.end_with?(gemfile)
+          Dir.mktmpdir do |dir|
+            gemfile = File.join(dir, gemfile)
+
+            File.write(gemfile, <<~GEMFILE)
+              source "https://rubygems.org"
+              git_source(:github) { |repo| "https://github.com/\#{repo}.git" }
+              #{rails_gemfile_entry}
+            GEMFILE
+
+            bundle_command("install", "BUNDLE_GEMFILE" => gemfile)
+
+            require "shellwords"
+            bundle_command("exec rails #{self_command} #{Shellwords.join(@argv)}", "BUNDLE_GEMFILE" => gemfile)
```

Comment:
I tried this commit in my machine and the fact that `"BUNDLE_GEMFILE"` is being set here is causing all the commands in the real generator to use this gemfile instead of the application's gemfile.

Reply:
Good catch!  I had tested this with `--skip-webpack-install` for performance reasons, and it worked, but I see now how it affects e.g. `rails webpacker:install`.  I can fix that, but instead I'm going to try a slightly different approach that doesn't require setting `BUNDLE_GEMFILE`.

[Application templates](https://guides.rubyonrails.org/rails_application_templates.html) will still be affected by the [`bundle exec`](https://github.com/rails/rails/pull/39663/files#diff-5762f42ce6ab7104d2d361e20112979aR443), but they can use an [inline gemfile](https://bundler.io/guides/bundler_in_a_single_file_ruby_script.html) to `require` third-party gems, if they need them.